pyam/units.py

```diff
@@ -0,0 +1,41 @@
+import copy
+import logging
+
+import pandas as pd
+import pint
+
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+# get application pint.UnitRegistry and load energy-units
+_REGISTRY = pint.get_application_registry()
+file = Path(__file__).parents[1] / 'energy-units' / 'definitions.txt'
+_REGISTRY.load_definitions(str(file))
+
+
+def convert_unit(df, current, to, factor=None, inplace=False):
+    """Internal implementation of unit conversion with explicit kwargs"""
+    ret = copy.deepcopy(df) if not inplace else df
```

Comment:
do we need `deepcopy` here, or can we use `df.copy()`?

Reply:
`df.copy()` is just a shim for `copy.deepcopy(df)` but I refactored all occurrences for nicer code...